org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/runner/statement/AssertNoRunningJobsStatement.java

```diff
@@ -65,37 +69,56 @@ public void evaluate() throws Throwable {
 
     final Set<IFuture<?>> scheduledFutures = jobListener.getScheduledFutures();
     if (!scheduledFutures.isEmpty()) {
-      assertNoRunningJobs(Jobs.newFutureFilterBuilder()
-          .andMatchFuture(scheduledFutures)
-          .toFilter());
+      assertNoRunningJobs(scheduledFutures);
     }
   }
 
   /**
    * Asserts that all jobs accepted by the given filter are in 'done' state.
    */
-  private void assertNoRunningJobs(final Predicate<IFuture<?>> jobFilter) {
+  private void assertNoRunningJobs(final Set<IFuture<?>> futures) {
     try {
+      //noinspection ResultOfMethodCallIgnored
       Thread.interrupted(); // clear the thread's interrupted status, in case the JUnit test interrupted the executing thread.
-      Jobs.getJobManager().awaitDone(jobFilter, AWAIT_DONE_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+      removeAndAwaitDone(futures);
     }
     catch (final TimedOutError e) { // NOSONAR
-      final List<String> runningJobs = findJobNames(jobFilter);
+      final List<String> runningJobs = futures.stream()
+          .filter(f -> !f.isDone()) // because of the TimedOutError there could still be done jobs in the list
+          .map(f -> f.getJobInput().getName())
+          .collect(Collectors.toList());
       if (!runningJobs.isEmpty()) {
         fail(String.format("Test failed because some jobs did not complete yet. [context=%s, jobs=%s]", m_context, runningJobs));
       }
     }
   }
 
   /**
-   * Finds running job names which comply with the given filter.
+   * Waits until the given futures are done and removes them.
    */
-  private List<String> findJobNames(final Predicate<IFuture<?>> filter) {
-    final List<String> jobs = new ArrayList<>();
-    for (final IFuture<?> future : Jobs.getJobManager().getFutures(filter)) {
-      jobs.add(future.getJobInput().getName());
+  private void removeAndAwaitDone(Set<IFuture<?>> futures) {
+    // implementation note: we do not use IJobManager.awaitDone because we want to report jobs running longer than 500ms.
```

Comment:
```suggestion
    // implementation note: we do not use IJobManager.awaitDone because we want to report jobs running longer than REPORT_THRESHOLD_NANOS
```